functions/userAuth.js

```diff
@@ -1,29 +1,71 @@
-const { ID } = require("appwrite");
-const sdk = require("node-appwrite")
+// Copyright (c) Gridiron Survivor.
+// Licensed under the MIT License.
 
-const user = async ({ req, res, log, error }) => {
+const { ID } = require('appwrite');
+const sdk = require('node-appwrite');
 
-    // Init SDK
-    const client = new sdk.Client();
+/**
+ * creates a new user record in the User collection
+ * @param param - The parameters
+ * @param param.req - The request object
+ * @param param.res - The response object
+ * @returns {void}
+ */
+const user = async ({ req, res }) => {
+  // Init SDK
+  const client = new sdk.Client();
 
-    const databases = new sdk.Databases(client);
+  const databases = new sdk.Databases(client);
 
-    client
-        .setEndpoint('https://cloud.appwrite.io/v1') // Your API Endpoint
-        .setProject(process.env.PROJECT_ID) // Your project ID
-        .setKey(process.env.X_Appwrite_Key) // Your secret API key
+  client
+    .setEndpoint('https://cloud.appwrite.io/v1') // Your API Endpoint
+    .setProject(process.env.PROJECT_ID) // Your project ID
+    .setKey(process.env.X_Appwrite_Key); // Your secret API key
 
-    // create a new user
-    if (req.method === "POST" && req.body.email && req.body.email && req.body['$id']) {
-        await databases.createDocument(process.env.DATABASE_ID, process.env.COLLECTION_USERS_ID, ID.unique(), {
-            "email": req.body.email,
-            "name": req.body.name,
-            "labels": req.body.labels,
-            "userId": req.body['$id']
-        });
-        return res.json({ msg: "User was created successfully!" });
-    }
+  // create a new user
+  if (
+    req.method === 'POST' &&
+    req.body.email &&
+    req.body.email &&
```

Comment:
Hey Chris! Will you please explain why `req.body.email` is used twice?

Reply:
I will leave this to @choir27 to explain since it's remaining from his code.

Reply:
I think this was supposed to be `req.body.name`, but i made the mistake of using `req.body.email` instead. @chris-nowicki you're more than welcome to change this if you need to.